src/jit/lower.cpp

```diff
@@ -3408,6 +3408,9 @@ bool Lowering::IndirsAreEquivalent(GenTreePtr candidate, GenTreePtr storeInd)
     if (pTreeA->OperGet() != pTreeB->OperGet())
         return false;
 
+    if (candidate->gtType != storeInd->gtType)
```

Comment:
OK, I used the following:

```C++
    assert(genActualType(candidate->gtType) == genActualType(storeInd->gtType));
    
    if (genTypeSize(candidate->gtType) != genTypeSize(storeInd->gtType))
        return false;
```
and crossgened mscorlib and other corefx assemblies (not all of them due to some issues with System.Runtime version). No diffs and no asserts were generated.